File: problems/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//Iniatializing custom functions
float count_letters(string text);
float count_words(string text);
float count_sentences(string text);
float coleman(float l, float w, float s);

// Alphabet symbols associated to Scrabble
char LETTERS[] = {'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'Y', 'Z'};
//Detecting the length of the array
int arr_len = sizeof(LETTERS) / sizeof(LETTERS[0]);


int main(void)
{
    string text = get_string("Text: ");
    int col = (int) coleman(count_letters(text), count_words(text), count_sentences(text));
    if (col > 16)
    {
        printf("Grade 16+\n");
    }
    else if (col < 0)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", col);
    }
}

//To calculate the amount of letters
float count_letters(string text)
{
    int letters, i, j, n;
    for (i = 0, letters = 0, n = strlen(text); i < n; i++)
    {
        for (j = 0; j < arr_len; j++)
        {
            if (toupper(text[i]) == LETTERS[j])
            {
                letters++;
            }
        }
    }
    return letters;
}

//To calculate the amount of words (separated by spaces and counting the last
//word when counter reach the last character)
float count_words(string text)
{
    int words, i, n;
    for (i = 0, words = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            words++;
        }
    }
    if (n == i && n > 0)
    {
        words++;
    }
    return words;
}

//To calculate sentences by dividing the text in its
//(., !, ?)
float count_sentences(string text)
{
    int sent, i, n;
    for (i = 0, sent = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sent++;
        }
    }
    return sent++;
}

//To calculate Coleman's formula considering
//results of operations with data types
float coleman(float l, float w, float s)
{
    float col = round(0.0588 * ((l / w) * 100.0) - 0.296 * ((s / w) * 100.0) - 15.8);
    return col;
}

File: problems/sentimental/readability.py
import cs50 as cs
import sys as s


# Alphabet symbols associated to Scrabble
LETTERS = 'ABCDEFGHIJKLMNOPQRSTUVWYZ'
# Detecting the length of the array

arr_len = len(LETTERS)


def main():
    text = cs.get_string("Text: ")
    col = coleman(count_letters(text), count_words(text), count_sentences(text))
    if col > 16:
        print("Grade 16+\n")
    elif col < 0:
        print("Before Grade 1\n")
    else:
        print(f"Grade {col}")

# To calculate the amount of letters


def count_letters(text):
    letters = 0
    n = len(text)
    for i in range(n):
        for j in range(arr_len):
            if text[i].upper() == LETTERS[j]:
                letters += 1
    # print(f"Letras: {letters}", end = " ")
    return letters

# To calculate the amount of words (separated by spaces and counting the last
# word when counter reach the last character)


def count_words(text):
    words, j = 0, 0
    n = len(text)
    for i in range(n):
        if text[i] == ' ':
            words += 1
        j += 1
    if n == j and n > 0:
        words += 1
    # print(f"Words: {words}", end = " ")
    return words

# To calculate sentences by dividing the text in its
# (., !, ?)


def count_sentences(text):
    sent = 0
    n = len(text)
    for i in range(n):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            sent += 1
    # print(f"Frases: {sent}", end = " ")
    return sent

# To calculate Coleman's formula considering
# results of operations with data types


def coleman(l, w, s):
    col = int(round(0.0588 * ((l / w) * 100.0) - 0.296 * ((s / w) * 100.0) - 15.8))
    return col


if __name__ == "__main__":
    main()